### DISCUSSION TITLE
Is the engine suitable for 3D games on PC?

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1449] [General]
I like the axmol interface and want to try my hand at 3D. Is the engine suitable for 3D games on PC? Or should we not look towards 3D games on this engine?

**Comment by DelinWorks** [comment]
If you're looking to make a game similar to minecraft or genshin impact type of graphics then yeah you will be able to pull it off with axmol, except for the ladder it might be harder, but if you want graphics like unreal engine then you will need an immense amount of work, axmol only provides basic if not excellent 3D functionality and excels at 2D so yea for games like minecraft and genshin (cartoon art) it's practical.
keep in mind that basic shaders need some work, like translucency, normal maps, and other PBR materials, also TAA but I think you can use MSAA which is built-in. that too with bone animations which I _really_ don't know about if they're implemented properly and that they might cause headaches.

**Reply by 10100010010001001010** [reply]
but what about performance?

**Reply by DelinWorks** [reply]
It greatly depends on the context and what you're trying to make, if you want to render too much grass then you should use Instanced Rendering which is already in the engine, and for other slow applications that can be implemented poorly with bad performance can instead be achieved by using features of hardware and specifications, OpenGL spec is modern and can be used for rendering with great performance, especially with the new shader compilation pipeline that halx has implementedm it would bring good performance because of uniform block objects, and uniformity compiling for different platforms, but keep in mind that you will not get as much performance as you would get with other APIs like Vulkan or DirectX, as for 2D Vulkan and DirectX are overkill, OpenGL is more than enough.
